pkgs/os-specific/linux/tuxedo-rs/default.nix

```diff
@@ -1,29 +1,43 @@
 { lib
 , fetchFromGitHub
 , rustPlatform
+, testers
+, tuxedo-rs
 }:
 let
 
+  name = "tuxedo-rs";
+  version = "0.3.0";
+  mainProgram = "tailor";
+  owner = "AaronErhardt";
+
+in
+rustPlatform.buildRustPackage {
+
+  pname = name;
+
+  inherit version;
+
   # NOTE: This src is shared with tailor-gui.
   # When updating, the tailor-gui.cargoDeps hash needs to be updated.
   src = fetchFromGitHub {
-    owner = "AaronErhardt";
-    repo = "tuxedo-rs";
-    rev = "74b863e6dcb1ec2e6c8fb02c16bb6f23b59e67f6";
-    hash = "sha256-Yujki2vGzaT8Ze5Usk8FPg8bn86MvyyPTiWuWwEw7Xs=";
+    inherit owner;
```

Comment:
It is generally undesired to use variables (besides version) to populate fetchFromGitHub because it makes overriding harder.

Reply:
question: do you have a link to a guideline for this?

I see

```nix
    repo = pname;
    rev = "v${version}";
```

in many packages, and typically overrides just override the entire `src` attrset (as the hash changes, too).

Reply:
Sadly, I don't have a guideline for this but I think I can explain why this is needed: If you call `overrideAttrs` on such a derivation trying to change `pname`, the fetcher would not be able to retrieve the sources anymore. However, sometimes you want to change the `pname` without changing the sources. For example, we do this for systemdMinimal.

Reply:
I see, thanks. Good to know. I suppose this isn't an issue for the version? In such a case, you would probably want the source to change with the version, right?

Reply:
Yes exactly!